src/main.ts

```diff
@@ -14,26 +14,55 @@ import fastifyStatic from '@fastify/static'
 import { join } from 'path'
 import cookie from '@fastify/cookie'
 import { PrismaExceptionFilter } from './exceptions/prisma-exception.filter'
+import { randomUUID } from 'crypto'
+import { requestContextStore } from './logging/request-context.service'
+import { CustomLogger } from './logging/custom-logger'
 
 const APP_LISTEN_CONFIG = {
   port: Number(process.env.PORT) || 3000,
   host: process.env.HOST || 'localhost',
 }
 
 const bootstrap = async () => {
+  const logger = new CustomLogger()
   const app = await NestFactory.create<NestFastifyApplication>(
     AppModule,
     new FastifyAdapter({
       ...(process.env.LOG_LEVEL
         ? {
+            disableRequestLogging: true,
             logger: { level: process.env.LOG_LEVEL },
+            genReqId: () => randomUUID(),
           }
         : {}),
     }),
     {
       rawBody: true,
+      logger,
     },
   )
+
+  app
+    .getHttpAdapter()
+    .getInstance()
+    .addHook('onRequest', (request, reply, done) => {
+      requestContextStore.run(request, () => {
+        logger.log('HTTP request received')
+        done()
```

Comment:
The context will be lost after done() is called since it's inside the callback. The request context won't be available for the rest of the request lifecycle. 

The proper pattern for Fastify hooks with AsyncLocalStorage is:

```
.addHook('onRequest', (request, reply, done) => {
  requestContextStore.enterWith(request)
  logger.log('HTTP request received')
  done()
})
```

Reply:
whoa, nice catch, will fix.